### PR TITLE
HIVE-24933: Replication fails for transactional tables having same na…

### PR DESCRIPTION
**Pull request by pmadhukar-cloudera**
…me as dropped non-transactional table

### CODE REVIEW

File: ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java [hunk 1]
```diff
@@ -1406,4 +1439,114 @@ public static Table tableIfExists(ImportTableDesc tblDesc, Hive db) throws HiveE
     }
   }
 
+  public static class LoadTableStateWrapper {
```
Comment (pkumarsinha) [inline ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java:1442]: Why are both LoadTableStateWrapper and DelayExecUtil needed? They are almost alike

File: ql/src/java/org/apache/hadoop/hive/ql/plan/CopyWork.java [hunk 2]
```diff
@@ -136,4 +140,18 @@ public boolean isOverwrite() {
...
+  public void setLoadTableStateWrapper(LoadTableStateWrapper loadTableStateWrapper) {
+    this.loadTableStateWrapper = loadTableStateWrapper;
+  }
+
+  public void setValuesBeforeExec() throws HiveException {
+    if (loadTableStateWrapper == null) {
+      return;
+    }
+
+    Table table = loadTableStateWrapper.getTableIfExists();
```
Comment (pkumarsinha) [inline ql/src/java/org/apache/hadoop/hive/ql/plan/CopyWork.java:153]: CopyWork was generic in nature. setValuesBeforeExec is tying it to resolve the path only from table. All we need is to be able to resolve the path dynamically. A resolver interface should be prefered.

File: ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java (hunk 1 above)
Comment (pmadhukar-cloudera) [reply ql/src/java/org/apache/hadoop/hive/ql/parse/ImportSemanticAnalyzer.java:1442]: DelayExecUtil has been renamed to LoadTableStateWrapper.

File: ql/src/java/org/apache/hadoop/hive/ql/plan/DeferredWorkHelper.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.hive.ql.plan;
+
+import org.apache.hadoop.hive.ql.metadata.HiveException;
+
+public interface DeferredWorkHelper {
+  void setupWork(CopyWork copyWork) throws HiveException;
```
Comment (pkumarsinha) [inline ql/src/java/org/apache/hadoop/hive/ql/plan/DeferredWorkHelper.java:24]: Can we not use generics?

**Pull request closed without merging**